test/testaudiocapture.c

```diff
@@ -199,38 +122,91 @@ int main(int argc, char **argv)
     device = SDL_OpenAudioDevice(want_device, NULL);
     if (!device) {
         SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't open an audio device for capture: %s!\n", SDL_GetError());
-        SDL_Quit();
-        exit(1);
+        return -1;
     }
     SDL_PauseAudioDevice(device);
     SDL_GetAudioDeviceFormat(device, &inspec, NULL);
     stream_in = SDL_CreateAudioStream(&inspec, &inspec);
     if (!stream_in) {
         SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create an audio stream for capture: %s!\n", SDL_GetError());
-        SDL_Quit();
-        exit(1);
+        return -1;
     } else if (SDL_BindAudioStream(device, stream_in) == -1) {
         SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't bind an audio stream for capture: %s!\n", SDL_GetError());
-        SDL_Quit();
-        exit(1);
+        return -1;
     }
 
     SDL_SetAudioStreamFormat(stream_in, NULL, &outspec);  /* make sure we output at the playback format. */
 
     SDL_Log("Ready! Hold down mouse or finger to record!\n");
 
-#ifdef __EMSCRIPTEN__
-    emscripten_set_main_loop(loop, 0, 1);
-#else
-    while (!done) {
-        loop();
-        if (!done) {
-            SDL_Delay(16);
+    return 0;
+}
+
+int SDL_AppEvent(const SDL_Event *event)
```

Comment:
Should this be SDL_bool? Are there different error codes here?

Reply:
-1 for "terminate the program with an error code", 1 for "terminate the program with a success code", and 0 to keep going (same as SDL_AppIterate).

I'll deal with the documentation shortly.

Reply:
Documentation is written in 4db2131589cff3c1820bfb8b1be55884c51191ae.